packages/spanner/src/database.js

```diff
@@ -713,45 +713,43 @@ Database.prototype.runStream = function(query, options) {
  * });
  *
  * //-
- * // If the callback is omitted, we'll return a Promise.
+ * // To use promises, simply return a Promise.
  * //-
- * database.runTransaction()
- *   .then(function(data) {
- *     var transaction = data[0];
- *
- *     // Run a transactional query.
- *     return transaction.run('SELECT * FROM Singers')
- *       .then(function() {
- *         // Queue a mutation (note there is no callback passed to `insert`).
- *         transaction.insert('Singers', {
- *           SingerId: 'Id3b',
- *           Name: 'Joe West'
- *         });
- *
- *         // Commit the transaction.
- *         return transaction.commit();
- *       });
- *   })
- *   .then(function() {
- *     // Transaction committed successfully.
+ * database.runTransaction(function(err, transaction) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // Run a transactional query.
+ *   return transaction.run('SELECT * FROM Singers').then(function() {
+ *     // Queue a mutation (note there is no callback passed to `insert`).
+ *     transaction.insert('Singers', {
+ *       SingerId: 'Id3b',
+ *       Name: 'Joe West'
+ *     });
+ *     // Commit the transaction.
+ *     return transaction.commit();
```

Comment:
> @lukesneeringer's statement is true, but also because we didn't want to mask anything to do with committing. We did this with Datastore, "just call done() when your transaction is ready to be committed" and it confused many users. We ended up exposing the explicit API, and haven't received any complaints.

I bow to user data. Something like what I showed above would work well in Python where context managers are a well-understood thing. But you are completely right that it is not nearly as prevalent of a pattern in JavaScript (and it is hard to do).